From 4f4c617237d0a5891410537ab770e7a0bc2b1652 Mon Sep 17 00:00:00 2001
From: Jaco Kroon <jaco@uls.co.za>
Date: Fri, 10 Jun 2022 13:07:37 +0200
Subject: [PATCH] Minimal PDE_DATA patch.

---
 include/dahdi/kernel.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/dahdi/kernel.h b/include/dahdi/kernel.h
index 22b9b66..98841a4 100644
--- a/include/dahdi/kernel.h
+++ b/include/dahdi/kernel.h
@@ -62,6 +62,10 @@
 #include <linux/pci-aspm.h>
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0)
+#define PDE_DATA(i)	pde_data(i)
+#endif
+
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 29)
 #define HAVE_NET_DEVICE_OPS
 #endif
-- 
2.35.1

